src/components/Conversation/Conversation.tsx

```diff
@@ -1,44 +1,34 @@
+import { Progress } from '@chakra-ui/react'
 import React, { useCallback, useEffect, useRef } from 'react'
-import Loader from '../../components/Loader'
 import useChat from '../../hooks/useChat'
-import useConversation from '../../hooks/useConversation'
+import useFetchMessages from '../../hooks/useFetchMessages'
+import useMessages from '../../hooks/useMessages'
+import useSendMessage from '../../hooks/useSendMessage'
 import MessageComposer from './MessageComposer'
 import MessagesList from './MessagesList'
 
 const Conversation = (): JSX.Element => {
   const { recipient } = useChat()
   const messagesEndRef = useRef(null)
 
-  const scrollToMessagesEndRef = useCallback(() => {
+  const scroll = useCallback(() => {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     ;(messagesEndRef.current as any)?.scrollIntoView({ behavior: 'smooth' })
   }, [])
 
-  const { messages, sendMessage, loading } = useConversation(
-    recipient,
-    scrollToMessagesEndRef
-  )
+  const messages = useMessages(recipient)
+  const { loading } = useFetchMessages(recipient, scroll)
+  const { send, loading: sending } = useSendMessage(recipient)
 
-  useEffect(() => {
-    if (messages.length === 0) return
-    scrollToMessagesEndRef()
-  }, [scrollToMessagesEndRef, recipient, messages])
+  useEffect(() => scroll, [scroll, recipient, messages])
```

Comment:
Do we need to call `scroll` here? `scroll()`

Reply:
This scroll is to scroll to the end of the page when the recipient changes or when there is a new message, this way new messages are displayed

Reply:
Don't think my comment was clear - I think we need **call** the function? Needs to be `scroll()` instead of just `scroll`?

Reply:
My bad I misread my own code...
This is super weird, this was working without calling it anyway should be fixed now